### PR TITLE
Implement sync status rpc

### PR DESCRIPTION
**Pull request by ercecan**
# Description
Implememnts syncing status rpc

### CODE REVIEW

File: crates/sequencer-client/src/lib.rs [hunk 1]
```diff
@@ -45,6 +45,12 @@ impl SequencerClient {
         }
     }
 
+    /// Gets l2 block height
+    #[instrument(level = "trace", skip(self), err, ret)]
+    pub async fn block_number(&self) -> Result<reth_primitives::U256, Error> {
```
Comment (kpp) [inline crates/sequencer-client/src/lib.rs:50]: Do we need this one?

File: crates/ethereum-rpc/src/lib.rs [hunk 2]
```diff
@@ -40,6 +40,34 @@ pub struct EthRpcConfig {
     pub eth_signer: DevSigner,
 }
 
+#[derive(
+    Clone,
+    Debug,
+    serde::Serialize,
+    serde::Deserialize,
+    PartialEq,
+    borsh::BorshDeserialize,
```
Comment (kpp) [inline crates/ethereum-rpc/src/lib.rs:49]: Do we need borsh?

File: crates/ethereum-rpc/src/lib.rs [hunk 3]
```diff
@@ -40,6 +40,34 @@ pub struct EthRpcConfig {
...
+    borsh::BorshSerialize,
+)]
+pub enum CitreaStatus {
+    Synced,
```
Comment (kpp) [inline crates/ethereum-rpc/src/lib.rs:67]: How about adding which L2 block N is synced?

File: crates/ethereum-rpc/src/lib.rs (hunk 2 above)
Comment (eyusufatik) [reply crates/ethereum-rpc/src/lib.rs:49]: yeah prob no lets remove

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: bin/citrea/tests/e2e/mod.rs [hunk 4]
```diff
@@ -3007,3 +3008,90 @@ async fn test_gas_limit_too_high() {
...
+            DEFAULT_MIN_SOFT_CONFIRMATIONS_PER_COMMITMENT,
+            true,
+            None,
+            None,
+            Some(true),
+            DEFAULT_DEPOSIT_MEMPOOL_FETCH_LIMIT,
+        )
+        .await;
+    });
+
```
[the remark this answers is not in the document]
Comment (ercecan) [reply bin/citrea/tests/e2e/mod.rs:3075]: I can get it up to 1000 to make it less flaky as possible

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**